### PR TITLE
Make Jackson + JSON-B @JsonSerializer/Deserializer for fields work in native

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java [hunk 1]
```diff
@@ -73,21 +78,40 @@ void register() {
...
+                AnnotationValue usingValue = deserializeInstance.value("using");
+                if (usingValue != null) {
+                    // the Deserializers are constructed internally by Jackson using a no-args constructor
+                    reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, usingValue.asClass().toString()));
+                }
+            }
+        }
+
+        // handle the various @JsonSerialize cases
+        for (AnnotationInstance serializeInstance : index.getAnnotations(JSON_SERIALIZE)) {
```
Comment (gsmet) [inline extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: Why do we only add the class hierarchy for reflection in the deserialize case?

File: integration-tests/pom.xml [hunk 2]
```diff
@@ -65,6 +65,7 @@
         <module>neo4j</module>
         <module>mongodb-client</module>
         <module>jackson</module>
+        <module>json</module>
```
Comment (gsmet) [inline integration-tests/pom.xml]: Please call it jsonb. And it needs to be added to the native builds pipeline, otherwise it won't be executed on CI.

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks mostly good. I added a couple of comments.

File: integration-tests/pom.xml (hunk 2 above)
Comment (geoand) [reply integration-tests/pom.xml]: I am surprised I didn't do this the first time around :)

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: We register the hierarchy when using the annotation on a field or method in either case.
The hierarchy comes into to play for deserialization when using `@JsonDeserialize` on a class which is a different case.
For the deserialization case we register the hierarchy when the annotation is applied to a Class.

Does that make sense?

File: integration-tests/pom.xml (hunk 2 above)
Comment (geoand) [reply integration-tests/pom.xml]: Fixed

**Review by gsmet — APPROVED** [review]

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java (hunk 1 above)
Comment (gsmet) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: I suppose so :)

Comment (geoand) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: I cleaned up what I wrote a little more for posterity :)

**Comment by gsmet** [comment]
CI failures are unrelated (Azure issues).

**Pull request merged**